core/src/main/java/fj/Equal.java

```diff
@@ -455,6 +442,21 @@ public boolean equal(A a1, A a2) {
     });
   }
 
+  /**
+   * An equal instance for the {@link Zipper} type.
+   *
+   * @param ea Equality across the elements of the zipper.
+   * @return An equal instance for the {@link Zipper} type.
+   */
+  public static <A> Equal<Zipper<A>> zipperEqual(final Equal<A> ea) {
+    Definition<A> eaDef = ea.def;
+    return equalDef((a1, a2) -> {
+      return Equal.equals0(Stream.class, a1.lefts(), a2.lefts(), () -> Equal.streamEqual(Equal.anyEqual())) &&
```

Comment:
I think `streamEqual` should use `Equal<A>` not `anyEqual`. also It would be best to save the `streamEqual` in a local variable for reuse.
Also `equals0`need to be used.

Reply:
partially updated. I'm not sure about ```equals0```